src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -42,14 +43,10 @@ def get_attrs(self, item_list, user, **kwargs):
             alert_rule_trigger__alert_rule_id__in=alert_rules.keys()
         ).exclude(sentry_app_config__isnull=True, sentry_app_id__isnull=True)
 
-        sentry_app_installations_by_sentry_app_id = (
-            SentryAppInstallation.objects.get_related_sentry_app_components(
-                organization_ids={
-                    alert_rule.organization_id for alert_rule in alert_rules.values()
-                },
-                sentry_app_ids=trigger_actions.values_list("sentry_app_id", flat=True),
-                type="alert-rule-action",
-            )
+        sentry_app_installations_by_sentry_app_id = app_service.get_related_sentry_app_components(
```

Comment:
Tight encapsulation of this into the control silo greatly reduces cross silo interactions.